docs/concept-workflows.md

```diff
@@ -1,30 +1,210 @@
+
+## Workflows have options
+
+When you start a Workflow, you can pass along parameters that tell the Temporal Server how to handle the Workflow.
+This includes the ability to set timeouts for Workflow execution, a retry policy, the Task Queue name, a data converter, and child Workflow options.
+
+### Timeout settings
+
+It's often necessary to limit the amount of time a specific Workflow can be running.
+
+To support this, the following three parameters can be provided to Workflow options:
+
+#### Execution timeout
+
+- **Description**: This is the maximum amount of time that a Workflow should be allowed to run including retries and any usage of the "Continue-as-new" feature.
+This is different from [Run timeout](#run-timeout).
+- **Use-case**:
+
+#### Run timeout
+
+- **Description**: This is the maximum amount of time that a single Workflow run is restricted to.
+- **Use-case**:
+
+#### Task timeout
+
+- **Description**: This is the maximum amount of time that the Server will wait for the Worker to start processing a Task after the Task has been pulled from the Task Queue.
+If a [Command Task](/docs/glossary/#command) is lost, it is retried after this timeout.
+- **Use-case**:
+
+### Retry policy
+
+There may be scenarios where you need to retry a Workflow's execution from the very beginning.
+In this case, you can use supply a retry policy when you start the Workflow.
+
+#### Initial interval
+
+- **Description**: Amount of time that must elapse before the first retry occurs.
+- **Use-case**:
+
+#### Backoff coefficient
+
+- **Description**: Retries occur exponentially.
+The coefficient specifies how fast the retry interval is growing. The coefficient of 1 means that the retry interval is always equal to the `InitialInterval`.
+- **Use-case**:
+
+#### Maximum interval
+
+- **Description**: Specifies the maximum interval between retries.
+Useful for coefficients of more than 1.
+- **Use-case**:
+
+#### Maximum attempts
+
+- **Description**: Specifies the maximum number of attempts that can be made to execute a Workflow in the presence of failures. If this limit is exceeded, the Workflow fails without retrying again.
+- **Use-case**:
+
+#### Non-retryable error reasons
+
+- **Description**: Specifies errors that shouldn't be retried.
+- **Use-case**: There may be errors that you know of that should not trigger a retry.
+In this case you can specify them such that if they occur, the Workflow will not be retried.
+
+### The Task Queue
+
+The only required Workflow options parameter is the name of a [Task Queue](/docs/concept-task-queues).
+Read the [Task Queues concept page](/docs/concept-task-queues) for a better overview.
+Essentially a Task Queue is the mechanism by which any given Worker knows which piece of code to execute next.
+A Workflow can only use one Task Queue, and the from a developer's perspective it is named and managed as a simple string value.
+
+### Custom Id
```

Comment:
WorkflowId

I don't think we have a concept of 'custom Id' anywhere in our docs or APIs.